File: Service/ClientCare/RMA.Service.ClientCare.Contracts/Entities/RolePlayer/RolePlayerGroupPolicy.cs
using System;
using System.Collections.Generic;

namespace RMA.Service.ClientCare.Contracts.Entities.RolePlayer
{
    public class RolePlayerGroupPolicy
    {
        public int CaseTypeId { get; set; }
        public string Code { get; set; }
        public int ParentPolicyId { get; set; }
        public string ParentPolicyNumber { get; set; }
        public DateTime PolicyInceptionDate { get; set; }
        public string ClientReference { get; set; }
        public string CompanyName { get; set; }
        public int ProductOptionId { get; set; }

        public RolePlayer MainMember { get; set; }
        public List<RolePlayer> Spouse { get; set; }
        public List<RolePlayer> Children { get; set; }
        public List<RolePlayer> ExtendedFamily { get; set; }
        public List<RolePlayer> Beneficiaries { get; set; }

    }
}

File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/models/role-player-group-policy.ts
import { RolePlayer } from './roleplayer';

export class RoleplayerGroupPolicy {
  code: string;
  caseTypeId: number;
  parentPolicyId: number;
  parentPolicyNumber: string;
  policyInceptionDate: Date;
  clientReference: string;
  companyName: string;
  productOptionId: number;

  mainMember: RolePlayer;
  spouse: RolePlayer;
  children: RolePlayer[];
  extendedFamily: RolePlayer[];
  beneficiaries: RolePlayer[];
}
